modules/drivers/snowflake/test/metabase/driver/snowflake_test.clj

```diff
@@ -1323,3 +1324,63 @@
+            query (-> (lib/query metadata-provider products)
+                      (lib/filter (lib/contains products-category "get" {:case-sensitive false}))
+                      (lib/fields [products-id products-name products-category])
+                      (lib/order-by products-id :asc)
+                      (lib/limit 3))]
+        (testing "returns correct rows"
+          (is (= [[1 "Rustic Paper Wallet" "Gizmo"]
+                  [2 "Small Marble Shoes" "Doohickey"]
+                  [3 "Synergistic Granite Chair" "Doohickey"]]
+                 (mt/rows (qp/process-query query)))))
+        (testing "generates correct SQL"
+          (is (str/includes? (:query (qp.compile/compile-with-inline-parameters query))
+                             "CONTAINS(LOWER(\"PUBLIC\".\"PRODUCTS\".\"CATEGORY\"), LOWER('get')")))))
+    (testing "starts-with"
+      (let [metadata-provider (mt/metadata-provider)
+            products (lib.metadata/table metadata-provider (mt/id :products))
+            products-category (lib.metadata/field metadata-provider (mt/id :products :category))
+            products-name (lib.metadata/field metadata-provider (mt/id :products :title))
+            products-id (lib.metadata/field metadata-provider (mt/id :products :id))
+            query (-> (lib/query metadata-provider products)
+                      (lib/filter (lib/starts-with products-category "Gad" {:case-sensitive false}))
+                      (lib/fields [products-id products-name products-category])
+                      (lib/order-by products-id :asc)
+                      (lib/limit 3))]
+        (testing "returns correct rows"
+          (is (= [[5 "Enormous Marble Wallet" "Gadget"]
+                  [11 "Ergonomic Silk Coat" "Gadget"]
+                  [16 "Fantastic Wool Shirt" "Gadget"]]
+                 (mt/rows (qp/process-query query)))))
+        (testing "generates correct SQL"
+          (is (str/includes? (:query (qp.compile/compile-with-inline-parameters query))
+                             "STARTSWITH(LOWER(\"PUBLIC\".\"PRODUCTS\".\"CATEGORY\"), LOWER('Gad')")))))
+    (testing "ends-with"
+      (let [metadata-provider (mt/metadata-provider)
+            products (lib.metadata/table metadata-provider (mt/id :products))
+            products-category (lib.metadata/field metadata-provider (mt/id :products :category))
+            products-name (lib.metadata/field metadata-provider (mt/id :products :title))
+            products-id (lib.metadata/field metadata-provider (mt/id :products :id))
+            query (-> (lib/query metadata-provider products)
+                      (lib/filter (lib/ends-with products-category "mo" {:case-sensitive false}))
+                      (lib/fields [products-id products-name products-category])
+                      (lib/order-by products-id :asc)
+                      (lib/limit 3))]
+        (testing "returns correct rows"
```

Comment:
No need to duplicate this setup 3x. May as well throw it in a function with pluggable `filter-fn` and so on. Likewise, let's check `:case-sensitive` at the same time. This should be possible to turn into a pretty compact series of test cases using some helpers, or even a `doseq`.